operators/src/engine/initialized_path.rs

```diff
@@ -0,0 +1,359 @@
+use async_trait::async_trait;
+
+use crate::{
+    error::Error,
+    util::{
+        input::{MultiRasterOrVectorOperator, RasterOrVectorOperator},
+        Result,
+    },
+};
+
+use super::{
+    ExecutionContext, InitializedRasterOperator, InitializedVectorOperator,
+    MultipleRasterOrSingleVectorSource, MultipleRasterSources, MultipleVectorSources,
+    SingleRasterOrVectorSource, SingleRasterSource, SingleVectorMultipleRasterSources,
+    SingleVectorSource,
+};
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Default)]
+pub struct WorkflowOperatorPath {
+    id: Vec<u8>,
+}
+
+impl WorkflowOperatorPath {
+    pub fn new(id: Vec<u8>) -> Self {
+        Self { id }
+    }
+
+    pub fn inner(self) -> Vec<u8> {
+        self.id
+    }
+
+    #[must_use]
+    pub fn clone_and_extend(&self, suffix: &[u8]) -> Self {
```

Comment:
why do you pass a `&[u8]` and not a `u8`? Do we need this?

An alternative could maybe be:
- A `child()` method that appends a `0` to the path
- A `sibling()` method that increments the current path index

Reply:
https://github.com/geo-engine/geoengine/pull/780/commits/adbf36471fce81e866e4720850f6fcee917fc4ae

I think the alternate solution could work but we can still change that later if it is really a problem.